src/custom/combinable_imports.rs

```diff
@@ -0,0 +1,103 @@
+use marker_api::{
+    ast::{AstPath, AstPathSegment},
+    prelude::*,
+    span::SpanSource,
+};
+
+marker_api::declare_lint! {
+    /// # What it does
+    /// TODO
+    COMBINABLE_IMPORTS,
+    Warn,
+}
+
+pub fn check_item<'ast>(cx: &'ast MarkerContext<'ast>, item: ast::ItemKind<'ast>) {
+    if let ast::ItemKind::Mod(module) = item
+        && !item.span().is_from_expansion()
```

Comment:
Don't we ignore macro-expanded code by default?

Reply:
Depends on what you mean with ignore. Lints emitted in macro generated code is still given to lint crates, it's just that lints are allow-by-default in them.

This is a habit from Clippy, where we this filter is not the default. There it's a really cheap check to return early, which is why it's often the first thing a lint does.